packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1090,6 +1094,216 @@ describe("SharedTree", () => {
+					const resubmitUndos = makeEdits(resubmitTree, resubmitPeer);
+
+					provider.processMessages();
+
+					if (otherPeer === "A" && resubmitPeer === "A") {
+						assert.deepEqual(
+							// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+							[...otherTree.flexTree.content.content.at(0)!.content],
+							["a", "b", "c", "b", "c"],
+						);
+						assert.deepEqual(
+							// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+							[...resubmitTree.flexTree.content.content.at(0)!.content],
+							["a", "b", "c", "b", "c"],
+						);
+					} else {
+						assert.deepEqual([...resubmitTree.flexTree.content.content], []);
+						assert.deepEqual([...otherTree.flexTree.content.content], []);
+					}
+
+					assert.equal(otherUndos[1].revert(), RevertibleResult.Success);
+					assert.equal(otherUndos[0].revert(), RevertibleResult.Success);
+					assert.equal(
+						resubmitUndos[resubmitPeer === "B" ? 1 : 0].revert(),
+						RevertibleResult.Success,
+					);
+					provider.processMessages();
+
+					// disconnect the resubmit tree
+					provider.trees[0].setConnected(false);
+
+					assert.equal(
+						resubmitUndos[resubmitPeer === "B" ? 0 : 1].revert(),
+						RevertibleResult.Success,
+					);
+					const otherTreeRoot = otherTree.flexTree.content.content.at(0);
+					if (otherTreeRoot !== undefined) {
+						assert.notDeepEqual([...otherTreeRoot.content], ["a"]);
+					}
+
+					provider.trees[0].setConnected(true);
+					provider.processMessages();
+
+					assert.deepEqual(
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						[...resubmitTree.flexTree.content.content.at(0)!.content],
+						["a"],
+					);
+					assert.deepEqual(
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						[...otherTree.flexTree.content.content.at(0)!.content],
+						["a"],
+					);
+				});
+			});
+
+			it("the restore of a tree edited on a branch", () => {
+				const provider = new TestTreeProviderLite(2);
+				const content = {
+					schema,
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: [["a"]],
+				};
+				const resubmitTree = provider.trees[0].schematizeInternal(content);
+				const { undoStack, unsubscribe } = createTestUndoRedoStacks(
+					resubmitTree.checkout.events,
+				);
+				const otherTree = provider.trees[1].schematizeInternal(content);
+				provider.processMessages();
+
+				// validate insertion
+				validateTreeContent(otherTree.checkout, content);
+
+				// fork and delete the tree on the original
+				const branch = resubmitTree.fork();
+
+				// edit the deleted tree on the branch
+				const outerList = branch.flexTree.content.content;
+				const innerList = (outerList.at(0) ?? assert.fail()).content;
+				innerList.insertAtEnd("b");
+
+				// disconnect the resubmit tree
+				provider.trees[0].setConnected(false);
+
+				resubmitTree.flexTree.content.content.removeAt(0);
```

Comment:
The comments above ("fork and delete the tree on the original" and "edit the deleted tree on the branch") would make more sense if this line were moved up right after `const branch = resubmitTree.fork();`